csrc/predicate_compute.cpp

```diff
@@ -491,16 +491,22 @@ namespace {
 
 // Select first warp of threads along TIDx axis and then use ptx::elect_sync
 // TODO If TIDx is known at compile-time, generate custom mask.
-Val* createElectSyncPredicate() {
+Val* createElectSyncPredicate(bool use_first_warp = true) {
   Val* warp_size = IrBuilder::create<Val>(32L, PrimDataType::UInt64);
   Val* full_mask_val = IrBuilder::create<Val>(0xFFFFFFFF, PrimDataType::UInt32);
   Val* elect_sync_val = IrBuilder::create<Val>(PrimDataType::Bool);
   IrBuilder::create<UnaryOp>(
       UnaryOpType::ElectSync, elect_sync_val, full_mask_val);
-  return SimplifyingIrBuilder::logicalAndExpr(
-      elect_sync_val,
-      IrBuilder::ltExpr(
-          NamedScalar::getParallelIndex(ParallelType::TIDx), warp_size));
+  // TIDx < 32 or TIDx >= blockDim.x - 32
```

Comment:
Why do we need the option to pick first or last warp?

Reply:
Added comments: 
```
 // If TIDx is used for both computation and TMA load, we should select a
  // thread from the last warp along TIDx.
```
The code structure is:
```
  if ((((nvfuser_index_t)threadIdx.x) >= 128)) {
    decreaseRegisters<120>();
    for(nvfuser_index_t i15 = 0; i15 < i0; ++i15) {
      if ((((Hopper::electSync(4294967295U) && (((nvfuser_index_t)threadIdx.x) >= (((nvfuser_index_t)blockDim.x) + -32))) && b9) && b10)) {
      }
    }
  }
```
Since the `electSync` is within the predicate `TIDx >= 128`, it can't select the first warp (TIDx=0~31) along TIDx.